components/proxy/src/main/java/com/hotels/styx/admin/handlers/MetricsHandler.java

```diff
@@ -39,6 +57,58 @@
      * @param cacheExpiration duration for which generated page content should be cached
      */
     public MetricsHandler(CodaHaleMetricRegistry metricRegistry, Optional<Duration> cacheExpiration) {
-        super(checkNotNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+        super(requireNonNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+
+        this.metricRegistry = metricRegistry;
+    }
+
+    @Override
+    public Observable<HttpResponse> handle(HttpRequest request) {
+        String path = removeFinalSlash(request.path());
+
+        if (path.equals("/admin/metrics")) {
+            return super.handle(request);
+        }
+
+        String metricName = metricNameFromPath(path);
+        boolean pretty = request.queryParam("pretty").isPresent();
```

Comment:
I wouldn't represent them as URL path components. This is because Coda Hale metric names are not really (counter-intuitively) scoped. 

For example `jvm.bufferpool.direct` is not really a scope in Coda Hale metrics. It can be simultaneously a prefix for other metrics like `jvm.bufferpool.direct.capacity` *and* a metric of its own.

Using path components to refer to a metric, as in `/metrics/jvm/bufferpool/direct`, gives an impression that they are scoped. Would this path refer to a metric of that name, or all metrics under that scope?

For that reason I propose to refer to the metrics names just as strings, so that `/metrics/jvm.bufferpool.direct` would be treated as all metrics that start with that name.